flag.go

```diff
@@ -0,0 +1,212 @@
+package jsonnext
+
+import (
+	"flag"
+	"fmt"
+)
+
+// ConfigFlags defines a set of flags in the given FlagSet for a Config struct
+// to populate its fields from the command line. The return value is a pointer
+// to the Config struct that stores the values of the flags.
+// The fields are populated with the following flags:
+//  Config.ImportPath:
+//   -J, -jpath
+//  Config.ExtVars:
+//   -V, -ext-str: ext var as string literal
+//   -ext-code: ext var as code literal
+//   -ext-str-file: ext var as string from file
+//   -ext-code-file: ext var as code from file
+//  Config.TLAVars:
+//   -A, -tla-str: top-level arg as string literal
+//   -tla-code: top-level arg as code literal
+//   -tla-str-file: top-level arg as string from file
+//   -tla-code-file: top-level arg as code from file
+func ConfigFlags(fs *flag.FlagSet) *Config {
+	c := NewConfig()
+	ConfigFlagsVar(fs, c)
+	return c
+}
+
+// ConfigFlagsVar defines a set of flags in the given FlagSet for a Config
+// struct to populate the fields from the command line. The argument c points
+// to the Config struct to populate. The set of flags defined is described in
+// the ConfigFlags function description.
+func ConfigFlagsVar(fs *flag.FlagSet, c *Config) {
+	StringSliceVar(fs, &c.ImportPath, "jpath", "Add a library search `dir`")
+	ExtStrVar(fs, c.ExtVars, "ext-str", "Add extVar `var[=str]` (from environment if <str> is omitted)")
+	ExtCodeVar(fs, c.ExtVars, "ext-code", "Add extVar `var[=code]` (from environment if <code> is omitted)")
+	ExtStrFileVar(fs, c.ExtVars, "ext-str-file", "Add extVar `var=file` string from a file")
+	ExtCodeFileVar(fs, c.ExtVars, "ext-code-file", "Add extVar `var=file` code from a file")
+
+	TLAStrVar(fs, c.TLAVars, "tla-str", "Add top-level arg `var=[=str]` (from environment if <str> is omitted)")
+	TLACodeVar(fs, c.TLAVars, "tla-code", "Add top-level arg `var[=code]` (from environment if <code> is omitted)")
+	TLAStrFileVar(fs, c.TLAVars, "tla-str-file", "Add top-level arg `var=file` string from a file")
+	TLACodeFileVar(fs, c.TLAVars, "tla-code-file", "Add top-level arg `var=file` code from a file")
+
+	// Add short flags. TODO(camh): consider making these optional.
```

Comment:
sorry not playing dumb here:
why would you want to make them optional?
because they might overload with other custom short flags?
how will that play with Kong?
fwiw, that todo confused me more than it enlightened me....

Reply:
My opinion is that a library should not squat on short flags - they belong to the application. but I wanted these as defaults, so I need to have some way to turn it off.

Reply:
will it work with kong?

Reply:
i dont know. You would have to define your own struct. Perhaps I should export the `vmVarMap` (and maybe add those constructors) so you can do your own kong struct with your own flag names. Makes sense. I'll do that in another commit though I think, when I address the todo in totality. It should not hurt as it will be exporting new stuff, not changing exported stuff.